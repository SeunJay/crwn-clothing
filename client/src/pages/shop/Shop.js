import React from "react";
import { Route } from "react-router-dom";
import { createStructuredSelector } from "reselect";
import { connect } from "react-redux";
import withSpinner from "../../components/with-spinner/WithSpinner";
import CollectionsOverview from "../../components/collections-overview/CollectionsOverview";
import CollectionPage from "../collection/Collection";
import { fetchCollectionsAsyncStart } from "../../redux/shop/shop.actions";
import { selectIsCollectionFetching, selectIsCollectionLoaded } from "../../redux/shop/shop.selector";

const CollectionsOverviewWithSpinner = withSpinner(CollectionsOverview);
const CollectionPageWithSpinner = withSpinner(CollectionPage);

console.log(CollectionsOverviewWithSpinner);

class Shop extends React.Component {
  async componentDidMount() {
    const { fetchCollectionsAsyncStart } = this.props;

    await fetchCollectionsAsyncStart();
  }

  render() {
    const { match, isCollectionFetching,isCollectionLoaded } = this.props;
    return (
      <div className="shop-page">
        <Route
          exact
          path={`${match.path}`}
          render={(props) => (
            <CollectionsOverviewWithSpinner
              isLoading={isCollectionFetching}
              {...props}
            />
          )}
        />
        <Route
          exact
          path={`${match.path}/:collectionId`}
          render={(props) => (
            <CollectionPageWithSpinner
              isLoading={!isCollectionLoaded}
              {...props}
            />
          )}
        />
      </div>
    );
  }
}

const mapStateToProps = createStructuredSelector({
  isCollectionFetching: selectIsCollectionFetching,
  isCollectionLoaded: selectIsCollectionLoaded,
});

const mapDispatchToProps = (dispatch) => ({
  fetchCollectionsAsyncStart: () =>
    dispatch(fetchCollectionsAsyncStart()),
});

export default connect(mapStateToProps, mapDispatchToProps)(Shop);
