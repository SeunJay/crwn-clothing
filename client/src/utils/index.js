import { applyMiddleware, createStore } from "redux";
import rootReducer from "../redux/root-reducer";
import { middlewares } from "../redux/store";

export const testStore = (initialState) => {
  const createStoreWithMiddleware = applyMiddleware(...middlewares)(
    createStore
  );
  return createStoreWithMiddleware(rootReducer, initialState);
};
