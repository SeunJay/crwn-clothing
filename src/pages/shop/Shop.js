import React from "react";
import { Route } from "react-router-dom";
import { createStructuredSelector } from "reselect";
import { connect } from "react-redux";
import withSpinner from "../../components/with-spinner/WithSpinner";
import CollectionsOverview from "../../components/collections-overview/CollectionsOverview";
import CollectionPage from "../collection/Collection";
import { fetchCollectionsAsyncStart } from "../../redux/shop/shop.actions";
import { selectIsCollectionFetching } from "../../redux/shop/shop.selector";

const CollectionsOverviewWithSpinner = withSpinner(CollectionsOverview);
const CollectionPageWithSpinner = withSpinner(CollectionPage);

console.log(CollectionsOverviewWithSpinner);

class Shop extends React.Component {
  async componentDidMount() {
    const { fetchCollectionsAsyncStart } = this.props;

    await fetchCollectionsAsyncStart();
  }

  render() {
    const { match, isCollectionFetching } = this.props;
    return (
      <div className="shop-page">
        <Route
          exact
          path={`${match.path}`}
          render={(props) => (
            <CollectionsOverviewWithSpinner
              isLoading={isCollectionFetching}
              {...props}
            />
          )}
        />
        <Route
          exact
          path={`${match.path}/:collectionId`}
          render={(props) => (
            <CollectionPageWithSpinner
              isLoading={isCollectionFetching}
              {...props}
            />
          )}
        />
      </div>
    );
  }
}

const mapStateToProps = createStructuredSelector({
  isCollectionFetching: selectIsCollectionFetching,
});

const mapDispatchToProps = (dispatch) => ({
  fetchCollectionsAsyncStart: () =>
    dispatch(fetchCollectionsAsyncStart()),
});

export default connect(mapStateToProps, mapDispatchToProps)(Shop);
