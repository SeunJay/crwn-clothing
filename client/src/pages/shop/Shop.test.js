import React from "react";
// import { Provider } from "react-redux";
import { shallow } from "enzyme";
import { testStore } from "../../utils";
import { Shop } from "./Shop";
// import configureStore from "redux-mock-store";

// const mockStore = configureStore();

const setUp = (initialState = {}) => {
  const props = {
    fetchCollectionsStart: jest.fn(),
  };
  const store = testStore(initialState);
  const enzymeWrapper = shallow(<Shop store={store} {...props} />);
  return {
    enzymeWrapper,
    store,
    props,
  };
};

describe("Shop Page", () => {
  let wrapper;

  let mockFetchCollectionsStart = jest.fn();

  beforeEach(() => {
    // const initialState = {
    //   shop: {
    //     isFetching: false,
    //   },
    // };

    // const mockProps = {
    //   fetchCollectionsStart: mockFetchCollectionsStart,
    // };

    // const { enzymeWrapper } = setUp(initialState);
    // wrapper = enzymeWrapper;
    wrapper = shallow(
      <Shop fetchCollectionsStart={mockFetchCollectionsStart} />
    );
  });

  it("should render the Checkout component", () => {
    expect(wrapper.length).toBe(1);
  });
});
