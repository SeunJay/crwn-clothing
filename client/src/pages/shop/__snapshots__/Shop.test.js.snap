// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shop Page take snapshot 1`] = `undefined`;
